src/CMakeLists.txt

```
@@ -12,452 +12,106 @@
 # *     - initial API and implementation and/or initial documentation
 # *******************************************************************************/
 
-set(FORTE_EXTERNAL_MODULES_DIRECTORY "" CACHE PATH "Path to a directory with FORTE modules located outside of the FORTE source code")
-SET(FORTE_MODULE_LIST modules;com;external)
-forte_create_modules_file("${FORTE_MODULE_LIST}")
-forte_add_subdirectory(arch)
-forte_add_subdirectory(core)
-foreach(FORTE_MODULE_DIR ${FORTE_MODULE_LIST})
-  string(COMPARE EQUAL "${FORTE_MODULE_DIR}" external _isExternal)
-  if(NOT _isExternal)
-    forte_add_subdirectory("${FORTE_MODULE_DIR}")
-  endif()
-endforeach(FORTE_MODULE_DIR)
-if(EXISTS ${FORTE_EXTERNAL_MODULES_DIRECTORY})
-  INCLUDE(${CMAKE_BINARY_DIR}/src/external/CMakeLists.txt)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+
+# FIXME: this is way too many
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/arch)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/core)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/core/cominfra)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/core/datatypes)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/core/datatypes/convert)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/arch/posix)
+
+target_compile_definitions(forte-global INTERFACE FORTE_${CMAKE_CXX_BYTE_ORDER})
+
+###############################################################################
+# Core infrastructure
+
+add_subdirectory(generated)
+
+if (FORTE_BUILD_EXECUTABLE)
+  add_executable(forte ${FORTE_GENERATED_FILES_DIR}/startuphook.cpp)
+  install(TARGETS forte RUNTIME DESTINATION bin)
```

Please also add it to an export set:
```suggestion
  install(TARGETS forte EXPORT forte-targets RUNTIME DESTINATION bin)
```

---

yes, definitely. I skipped that for now, because that area has some additional discussion points. My goal is definitely to have a standard (modern) cmake package file.